### DISCUSSION TITLE
Default Sound device

### DISCUSSION THREAD

**Discussion by callagg2** [linuxmint/discussions#61] [Ideas]
In Linux Mint's Sound Applet, I don't seem to have the option to "easily" select a default sound output.

I currently have three sound outputs:

1. Analog Output Logitech Wireless Headset (connected via USB dongle)
2. HDMI/DisplayPort 2 Built-in Audio
3. Line Out Built-in Audio

While I wanted the HDMI to be the default, 
because the USB dongle of the wireless headset was plugged in, it would always default to that. 

The solution involved the following steps:
sudo xed /etc/pulse/default.pa

Then add the following lines to the file:
### Make some devices default
#set-default-sink output
#set-default-source input
set-default-sink alsa_output.pci-0000_00_1b.0.hdmi-stereo-extra1
set-card-profile alsa_card.pci-0000_00_1b.0 output:hdmi-stereo+input:analog-stereo
 
Whereas in Windows, I can just right click inside the Sound applet and choose the default device.
Maybe Linux Mint could look at simplifying the default option.

**Comment by kacperpaczos** [comment]
I have noticed another issue with the sound on my laptop.

When I use Bluetooth headphones and the connection is interrupted, the sound settings remain the same, so I don't have any sound. I have to manually go back to the previous speaker settings on my laptop in order to play sounds. It should happen automatically.

I believe the sound subsystem should stop playing music if I change the sound output device (headphones, speakers). Android does something similar to that.

**Comment by claudiux** [comment]
@callagg2 Right-click on the sound applet and select "Output device". Choose your device from the drop-down list that has just opened.

**Reply by callagg2** [reply]
Claudiux, 
you're quite right you can do that every single time you start up your computer,
but I wanted to be able to set a particular default output once
and the sound output only change to another output when i manually change it.
But as things stand, my computer constantly defaults to the USB wireless headset,
unless i change the file via sudo xed /etc/pulse/default.pa

**Comment by claudiux** [comment]
@callagg2 
Please take a look at the `~/.config/pulse/*default*` files. Are you satisfied with their contents?

Compare them with the output of these commands:

  pacmd list-sinks
  pacmd list-sources

**Reply by callagg2** [reply]
Yes, when i run the command "xed ~/.config/pulse/*default*" it opens a few xed documents.

The first being an default-sink  file
with the correct name alsa_output.pci-0000_00_1b.0.hdmi-stereo-extra1
the second being a default-source file
with the correct name alsa_input.usb-Logitech_Logitech_Wireless_Headset-00.mono-fallback

This matches the names in pacmd list-sinks 
(some extracts)

index: 0
name: <alsa_output.usb-Logitech_Logitech_Wireless_Headset-00.analog-stereo>
card: 0 <alsa_card.usb-Logitech_Logitech_Wireless_Headset-00>
ports:
analog-output: Analog Output (priority 9900, latency offset 0 usec, available: unknown)
properties:
active port: <analog-output>


* index: 2
name: <alsa_output.pci-0000_00_1b.0.hdmi-stereo-extra1>
card: 1 <alsa_card.pci-0000_00_1b.0>
ports:
hdmi-output-1: HDMI / DisplayPort 2 (priority 5800, latency offset 0 usec, available: yes)
active port: <hdmi-output-1>

and under pacmd list-sources

index: 0
name: <alsa_output.usb-Logitech_Logitech_Wireless_Headset-00.analog-stereo.monitor>
card: 0 <alsa_card.usb-Logitech_Logitech_Wireless_Headset-00>

* index: 1
name: <alsa_input.usb-Logitech_Logitech_Wireless_Headset-00.mono-fallback>
card: 0 <alsa_card.usb-Logitech_Logitech_Wireless_Headset-00>
ports:
analog-input-mic: Microphone (priority 8700, latency offset 0 usec, available: unknown)
properties:
device.icon_name = "audio-input-microphone"
active port: <analog-input-mic>

 index: 3
name: <alsa_input.pci-0000_00_1b.0.analog-stereo>
card: 1 <alsa_card.pci-0000_00_1b.0>
ports:
analog-input-internal-mic: Internal Microphone (priority 8900, latency offset 0 usec, available: unknown)
properties:
device.icon_name = "audio-input-microphone"
analog-input-dock-mic: Dock Microphone (priority 7800, latency offset 0 usec, available: no)
properties:
device.icon_name = "audio-input-microphone"
analog-input-mic: Microphone (priority 8700, latency offset 0 usec, available: no)
properties:
device.icon_name = "audio-input-microphone"
active port: <analog-input-internal-mic>

index: 4
name: <alsa_output.pci-0000_00_1b.0.hdmi-stereo-extra1.monitor>
card: 1 <alsa_card.pci-0000_00_1b.0>
